File: backend/src/main/java/com/airassist/backend/dto/user/UserDTO.java
package com.airassist.backend.dto.user;

import com.airassist.backend.model.Roles;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class UserDTO {

    @Email(message = "Email must be valid")
    @Size(max = 254, message = "Email must be less than 254 characters")
    private String email;

    @Size(min = 6, max = 20, message = "Password must be between 6 and 20 characters")
    private String password;

    @Size(max = 50, message = "First name must be less than 50 characters")
    private String firstName;

    @Size(max = 50, message = "Last name must be less than 50 characters")
    private String lastName;

    @Enumerated(EnumType.STRING)
    private Roles role;

    private UserDetailsDTO userDetails;

}

File: frontend/src/app/shared/dto/user.dto.ts
import { Role } from '../enums/role.enum';
import { CaseDTO } from './case.dto';
export type UserDTO = {
  id?: string;
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  role: Role;
  userDetails: {
    id?: string;
    phoneNumber: string;
    address: string;
    postalCode: string;
    birthDate: Date;
  };
  comments?: Comment[];
  cases?: CaseDTO[];
  isFirstLogin?: boolean;
};
